docs/shared-asyncapi.md

```diff
@@ -2,128 +2,94 @@
-* Definitions for Models classes are generated from code and annotations.
-* This module is used to combine the models and template to a self-contained spec file.
-* The generated AsyncAPI spec is committed into source control.
-
-A manual written AsyncAPI spec template might look like this and can be stored as a resource:
-
+The suggested way to use this module is:
+
+- A template file defining the general `info`rmation, `channels` and `components.messages` using
+  the AsyncAPI spec.
+  `components.schemas` should be omitted.
+- The schema is defined and documented as Java classes in the code as they are used in message
+  handlers and consumers.
+  Jackson, Jakarta Validation and Swagger 2 annotations can be used for documentation.
+- The root classes of messages are referenced in `components.messages.YourMessage.payload.$ref` as
+  `class://your.package.MessageModel`.
+- The [`AsyncApiGenerator`](https://github.com/SDA-SE/sda-dropwizard-commons/blob/master/sda-commons-shared-asyncapi/src/main/java/org/sdase/commons/shared/asyncapi/AsyncApiGenerator.java)
+  is used to combine the template and the generated Json Schema of the models to a self-contained
+  spec file.
+- The generated AsyncAPI spec is committed into source control.
+  This way, the commit history will show intended and unintended changes to the API and the API spec
+  is accessible any time without executing any code.
+- The API can be view in [AsyncAPI Studio](https://studio.asyncapi.com/).
+
+It is suggested to use it as a test dependency, build the AsyncAPI in a unit test and verify that it
+is up-to-date.
+The [`GoldenFileAssertions`](https://github.com/SDA-SE/sda-dropwizard-commons/blob/master/sda-commons-server-testing/src/main/java/org/sdase/commons/server/testing/GoldenFileAssertions.java)
+from the test module help here.
+
+!!! example "Example: Build AsyncAPI for Cars"
+=== "asyncapi_template.yaml"
 ```yaml
-asyncapi: '2.3.0'
-id: 'urn:org:sdase:example:cars'
-defaultContentType: application/json
-
-info:
-  title: Cars Example
-  description: This example demonstrates how to define events around *cars*.
-  version: '1.0.0'
-
-channels:
-  'car-events':
-    publish:
-      operationId: publishCarEvents
-      summary: Car related events
-      description: These are all events that are related to a car
-      message:
-        oneOf:
-          - $ref: '#/components/messages/CarManufactured'
-          - $ref: '#/components/messages/CarScrapped'
-
-components:
-  messages:
-    CarManufactured:
-      title: Car Manufactured
-      description: An event that represents when a new car is manufactured
-      payload:
-        $ref: './schema.json#/definitions/CarManufactured'
-    CarScrapped:
-      title: Car Scrapped
-      description: An event that represents when a car is scrapped
-      payload:
-        $ref: './schema.json#/definitions/CarScrapped'
+--8<-- "sda-commons-asyncapi/src/test/resources/demo/asyncapi_template.yaml"
```

Comment:
The examples are missing some space indents.

4 spaces for all for the notes (below the title with !!! or ???) and additional 4 spaces for the tab (the ones with ===) content. I guess, something got lost on copy paste from [the original docs](https://github.com/SDA-SE/sda-spring-boot-commons/blob/master/docs/asyncapi.md?plain=1#L38-L58).